shared/src/commonMain/kotlin/org/hyperskill/app/study_plan/widget/presentation/StudyPlanWidgetReducer.kt

```diff
@@ -82,13 +81,22 @@ class StudyPlanWidgetReducer : StateReducer<State, Message, Action> {
             )
         }
 
+        val sortedVisibleSections =
+            if (state.studyPlan != null) {
+                val visibleSectionsIds = visibleSections.map { it.id }.toSet()
+                state.studyPlan.sections.intersect(visibleSectionsIds).toList()
+            } else {
+                return state to emptySet()
+            }
+
         val loadedSectionsState = state.copy(
+            studyPlan = state.studyPlan.copy(sections = sortedVisibleSections),
```

Comment:
Modifying of the study plan sections state for preserving sort order confuses me 🤔 

Do we have a place where we are getting sections from a remote data source by `studyPlan.sections` ids, I mean could we meet a bug when a section `isVisible` value can change in the backend, but we will not update it, because `state.studyPlan.sections` doesn't contain it anymore?

Reply:
We just want to display visible sections reordered on study plan sections

Reply:
> could we meet a bug when a section isVisible value can change in the backend, but we will not update it, because state.studyPlan.sections doesn't contain it anymore?

All the data is reloading on every screen showing so the study plan is loading too.

Reply:
@ivan-magda 
But anyway, wee we can still original `StudyPlan.sections`.
Soring is not relay on it.